### PR TITLE
Groundwork luluc (Biomee)

### PR DESCRIPTION
**Pull request by marcadella**
- Reworked internal data structures as groundwork for LULUC.
- Only external change is the addition of `lu_fraction` in the output (non-breaking as it is appended in last position).
- Fixed a bug where cohort output were not properly numbered for multi-year simulation.

### CODE REVIEW

**Review by fabern — COMMENTED** [review]
Thanks @marcadella for this groundwork.

As discussed over lunch, I think we should clarify at what level in the code we are dealing with the land use types (currently 4 of them) and at what level we are dealing with the land units (can be many more since we have multiple instances of secondary forest). I suggest to talk about these naming conventions (and output) in a common meeting with everybody involved. (Easier than online.)

I'd argue that the naming in the code should be modified to facilitate this distinction. I.e. we should refrain from using `_lu` for variable naming, but rather favor `_tile` or `_landuse` if needed.

**Comment by stineb** [comment]
I think it would be best if @marcadella advances the development of this branch, we give inputs if he asks for it, and we then take stock together once the first milestone is achieved. @marcadella please let us know when you're there.

In case it's urgent:

`lu`: refers to land unit, used for example as the index of the tile (=land unit) dimension. Respective outputs should be given for each `lu`.

**Comment by fabern** [comment]
@stineb: am I understanding this correctly that land unit `lu` then can be up to e.g. 50 different tiles at each moment in time? Furthermore, with land use change they can are created and merged, thus the output on tile level would be very unstructured. 

Wouldn't output on the aggregated land use type level (as suggested by marcadella) be sufficient?

**Comment by marcadella** [comment]
@stineb The reason I am pushing to master is because Fabian an myself are both working on the same pieces of code. If I code all LULUC and merge in one month time, the discrepancy between Fabian's codebase and mine will be so great that it will be a nightmare to merge. Therefore I prefer to push often to master (I always ensure that the code is backward compatible and all the tests are passing so my changes basically do not alter the programs output).

@fabern You are mixing up cohort and land unit. Each land unit (LU) is a  BiomeE tile associated with a real number specifying the area fraction occupied by this LU, and a definition of the land use that applies on this tile. Each tile contains up to 50 cohorts, but the concept of cohort is an implementation detail of a Biomee tile which LULUC does not care about (except during initialization where the user defines which PFT grows on which LU).

File: R/run_biomee_f_bysite.R [hunk 1]
```diff
@@ -252,11 +252,26 @@ run_biomee_f_bysite <- function(
...
 ){
-  
-  # predefine variables for CRAN check compliance
-  type <- NULL
+
+  # If init_lu is null, we create dummy LU initial state containing only one state (with a fraction of 1)
+  if (is.null(init_lu))
+    init_lu <- c(1.0)
+
+  # Number of LU states
```
[the remark this answers is not in the document]
Comment (marcadella) [reply R/run_biomee_f_bysite.R:264]: Idem.

File: R/data.R [hunk 2]
```diff
@@ -356,16 +356,14 @@
...
-#' See \code{\link{run_biomee_f_bysite}} for a detailed 
-#' description of the outputs.
-"biomee_p_model_output"
-
 #' rsofun BiomeE (gs_leuning) output data
 #'
-#' Example output dataset from a BiomeE-model run (gs_leuning)
-#' See \code{\link{run_biomee_f_bysite}} for a detailed 
-#' description of the outputs.
+#' Example output dataset from a BiomeE-model run using divers \code{\link{biomee_p_model_drivers}}
```
[the remark this answers is not in the document]
Comment (marcadella) [reply R/data.R:361]: Thanks. Fixed on my branch.

File: R/run_biomee_f_bysite.R [hunk 3]
```diff
@@ -5,21 +5,21 @@
 #' @param sitename Site name.
 #' @param params_siml Simulation parameters.
 #' @param site_info Site meta info in a data.frame.
-#' @param forcing A data frame of forcing climate data, used as input.
+#' @param forcing A data.frame of forcing climate data, used as input.
+#' @param init_lu Initial land use state array used as input (optional).
+#' @param luc Land use change (transition matrix) data.frame used as input (optional).
```
[the remark this answers is not in the document]
Comment (marcadella) [reply R/run_biomee_f_bysite.R:10]: You're right, I commented out on my branch.

**Comment by fabern** [comment]
No, I'm currently not working on any BiomeE updates. 
So I don't expect any divergences in the coming month.
PR frequency is thus a matter of style/choice. 

I'm not sure I understand then the concept of LU. 
I am however positive, that I'm not confounding LU with cohorts:

Last time we discussed that each tile has a history. 
And that there can be multiple secondary forest tiles, each with a distinct history. 
Basically an additional secondary forest tile gets created whenever managed land gets abandoned. This can happen once in 2010, then another time in 2015, creating thus 2 tiles. Unless we merge all of them together, this creates a variable number of tiles. Is this land use change not part of that PR?

I am confused now about the objective of this PR.

File: R/run_biomee_f_bysite.R [hunk 4]
```diff
@@ -5,21 +5,21 @@
...
 #' @param params_tile Tile-level model parameters, into a single row data.frame.
 #' @param params_species A data.frame containing species-specific model parameters,
-#'   with one species per row. See examples \code{\link{biomee_gs_leuning_drivers}} or \code{\link{biomee_p_model_drivers}}
+#'   with one species per row.
```
[the remark this answers is not in the document]
Comment (marcadella) [reply R/run_biomee_f_bysite.R:13]: Yes, not sure why this one is gone. Added to my branch again.

**Comment by marcadella** [comment]
Merged in branch luluc.

**Comment by fabern** [comment]
:raised_hands:
